data/mods/gen7mixandmega/scripts.ts

```diff
@@ -99,10 +99,11 @@ export const Scripts: ModdedBattleScriptsData = {
 				species.types = [species.types[0], deltas.type];
 			}
 			const baseStats = species.baseStats;
-			for (const statName in baseStats) {
+			for (const statName of Stats.statIDs) {
 				baseStats[statName] = this.battle.clampIntRange(baseStats[statName] + deltas.baseStats[statName], 1, 255);
 			}
 			species.weighthg = Math.max(1, species.weighthg + deltas.weighthg);
+			// @ts-ignore
```

Comment:
ts-ignores should explain why they are necessary

Reply:
Added an explanation. Lmk if it's preferable to do `(species as any).originalMega` instead, in this case, which would be more consistent with how it's done in the rest of the codebase when adding a property to a class.

On a side note, I wonder if it would be helpful to add something like a `MixedSpecies` type, which would extend `Species` with the `originalMega` property. We could probably have a generic `AugmentedSpecies<T>` class, where `MixedSpecies = AugmentedSpecies<{ originalMega: ___ }>`. Then `doGetMixedSpecies` would return that `AugmentedSpecies` class instead, so callers of this function know that it returns something with the `originalMega` property.